### PR TITLE
Fix tests failing with subprocess aborted error [16199]

### PR DESCRIPTION
**Pull request by rsanchez15**
This PR intends to fix the failing tests issued at compiling Fast DDS with statistics module enabled. 

The test not covered by this PR, and so they continue failing are:

- PubSub layer (deprecated):
  - BlackboxTests_FastRTPS.PubSubHistory.ReliableTransientLocalKeepLast1Data300Kb.Transport (Timeout)
- Statistics (needs some re-design):
  - BlackboxTests_DDS_PIM.PubSubBasic.ReceivedPropertiesDataWithinSizeLimit.Transport (Failed)
  - BlackboxTests_DDS_PIM.PubSubBasic.ReceivedPropertiesDataWithinSizeLimit.Intraprocess (Failed)
  - BlackboxTests_DDS_PIM.PubSubBasic.ReceivedPropertiesDataWithinSizeLimit.Datasharing (Failed)
  - UDPv4Tests.send_to_wrong_interface (Subprocess aborted)
  - UDPv6Tests.send_to_wrong_interface (Subprocess aborted)
- Others:
  - BlackboxTests_FastRTPS.TransportTCP.TCP_TLS.Transport (Timeout)
  - BlackboxTests_DDS_PIM.TransportTCP.TCP_TLS.Transport (Timeout)
- Flaky tests: 
  - BlackboxTests_DDS_PIM.DDSDiscovery.AddDiscoveryServerToList (Failed)

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by rsanchez15** [comment]
@richiprosima please test this

**Comment by rsanchez15** [comment]
@richiprosima please test this

**Comment by rsanchez15** [comment]
@richiprosima please test this

File: test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp [hunk 1]
```diff
@@ -70,6 +70,48 @@ static const std::map<std::string, std::set<std::string>> type_compatibility_mat
     {"ENUM2_STR", {"ENUM2", "CHAR", "STRING"}}
 };
 
+static const std::vector<std::string> operators
```
Comment (jparisu) [inline test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp:73]: Is static required here?
Also, I would encapsulate those elements that are only for test propose in `test` namespace

**Review by jparisu — CHANGES_REQUESTED** [review]
Please check the following test, that did not fail before.
If it is flaky, it should be labeled as so.

845 | BlackboxTests_DDS_PIM.DDSStatus.IncompatibleQosConditions.Datasharing

File: test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp (hunk 1 above)
Comment (jparisu) [reply test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp:73]: Refactor for other PR

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**